internal/engine/build_and_deployer.go

```diff
@@ -52,6 +55,26 @@ func NewCompositeBuildAndDeployer(builders BuildOrder, shouldFallBack FallbackTe
 
 func (composite *CompositeBuildAndDeployer) BuildAndDeploy(ctx context.Context, manifest model.Manifest, currentState BuildState) (BuildResult, error) {
 	var lastErr error
+	changedConfigFiles := getChangedConfigFiles(currentState.FilesChanged(), manifest)
+	if len(changedConfigFiles) > 0 {
+		logger.Get(ctx).Verbosef("Detected a config file change (%v), re-executing tiltfile and doing an image build", changedConfigFiles)
+		lastBuilder := composite.builders[len(composite.builders)-1]
+		configChangeState := currentState.NewStateWithConfigFilesRemoved(manifest.ConfigMatcher)
```

Comment:
this doesn't seem right. it seems like you have to fall back to a BuildStateClean, because you don't know what's changed?